packages/components/src/disclosure/index.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { DisclosureContentProps } from './types';
+import type { WordPressComponentProps } from '../context';
+
+/**
+ * Accessible Disclosure component that controls visibility of a section of
+ * content. It follows the WAI-ARIA Disclosure Pattern.
+ *
+ * @see https://ariakit.org/reference/disclosure-content
+ *
+ * The plan is to build own API that accounts for future breaking changes
+ * in Ariakit (https://github.com/WordPress/gutenberg/pull/28085).
```

Comment:
I also wonder if this comment is necessary?

Reply:
Removed via https://github.com/WordPress/gutenberg/pull/55639/commits/f59bcf3878b27626232abe3c05c5dc512692ed47